src/test/ui/array_const_index-0.stderr

```diff
@@ -8,5 +8,12 @@ LL | const B: i32 = (&A)[1];
    |
    = note: #[deny(const_err)] on by default
 
-error: aborting due to previous error
+error[E0080]: erroneous constant used
+  --> $DIR/array_const_index-0.rs:17:13
+   |
+LL |     let _ = B; //~ ERROR erroneous constant used
+   |             ^ referenced constant has errors
```

Comment:
So we now error at definition site *and* at use site, but the former is "just" an err-by-default lint? This still complains about the same problem twice, but as long as it's just a lint at definition site I guess this is hard to avoid.

What I do not understand is why behavior changed here, why we now have a hard error in addition to the lint.

Reply:
If you allow the warning, you get an error. All that happened is that the error is reported a little earlier, before the compiler decides to stop compiling further due to "n previous errors"